zipkin-lens/src/components/TraceViewer/index.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import Dropzone from 'react-dropzone';
+
+import DetailedTraceSummary from '../DetailedTraceSummary';
+import { treeCorrectedForClockSkew, detailedTraceSummary } from '../../zipkin';
+
+const dropzoneBaseStyle = {
+  width: 200,
+  height: 60,
+  fontSize: '18px',
+  padding: '8px',
+  borderWidth: 2,
+  borderColor: '#666',
+  borderStyle: 'dashed',
+  borderRadius: 5,
+};
+
+const dropzoneActiveStyle = {
+  borderStyle: 'solid',
+  borderColor: '#6c6',
+  backgroundColor: '#eee',
+};
+
+class TraceViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startTs: null,
+      endTs: null,
+      traceSummary: null,
+      cannotOpenFile: false,
+      isInvalidFormat: false,
+    };
+    this.handleDrop = this.handleDrop.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleStartAndEndTsChange = this.handleStartAndEndTsChange.bind(this);
+  }
+
+  handleDrop(files) {
+    const [file] = files;
+    const fileReader = new FileReader();
+
+    fileReader.onload = () => {
+      const { result } = fileReader;
+      let rawTrace;
+      try {
+        rawTrace = JSON.parse(result);
+      } catch (error) {
+        this.setState({
+          traceSummary: null,
+          cannotOpenFile: false,
+          isInvalidFormat: true,
+        });
+      }
+      const traceSummary = detailedTraceSummary(
+        treeCorrectedForClockSkew(rawTrace),
+      );
+
+      this.setState({
+        startTs: 0,
+        endTs: traceSummary.duration,
+        traceSummary,
+        cannotOpenFile: false,
+        isInvalidFormat: false,
+      });
+    };
+
+    fileReader.onabort = () => {
+      this.setState({
+        traceSummary: null,
+        cannotOpenFile: true,
+        isInvalidFormat: false,
+      });
+    };
+    fileReader.onerror = fileReader.onabort;
+    fileReader.readAsText(file);
+  }
+
+  handleCancel() {
+    this.setState({
+      traceSummary: null,
+      cannotOpenFile: false,
+      isInvalidFormat: false,
+    });
+  }
+
+  handleStartAndEndTsChange(startTs, endTs) {
+    this.setState({ startTs, endTs });
+  }
+
+  renderDropzone() {
+    return (
+      <Dropzone
+        onDrop={this.handleDrop}
+        onFileDialogCancel={this.handleCancel}
+      >
+        {({ getRootProps, getInputProps, isDragActive }) => {
+          const styles = isDragActive
+            ? { ...dropzoneBaseStyle, dropzoneActiveStyle }
+            : { ...dropzoneBaseStyle };
+          return (
+            <div
+              {...getRootProps()}
+              style={styles}
+            >
+              <input {...getInputProps()} />
+              <p>Choose file</p>
```

Comment:
I'm not sure the right verbosity here, but I think we should guide users more. The classic UI has a "Browse" button at least, which I think gives more an indication that clicking it will allow one to select a file from their computer to upload. An improvement on that might be to reword with something like "Upload a file by drag/drop or click to browse".
As an aside, we haven't gotten any reports of confusion, but maybe even more explanation somehow about what kind of file/contents are expected would be good. Anyways, that could be a separate discussion from this pull request.